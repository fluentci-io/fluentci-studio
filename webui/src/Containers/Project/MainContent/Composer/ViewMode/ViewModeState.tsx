import { atom } from "recoil";

export const ViewModeState = atom<"stacked" | "code">({
  key: "ViewModeState",
  default: "stacked",
});
