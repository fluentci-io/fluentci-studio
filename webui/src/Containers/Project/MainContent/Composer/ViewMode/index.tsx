import ViewMode from "./ViewModeWithData";

export default ViewMode;
