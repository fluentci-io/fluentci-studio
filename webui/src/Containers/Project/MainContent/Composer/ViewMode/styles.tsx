import { css } from "@emotion/react";
import styled from "@emotion/styled";

export const Container = styled.div`
  position: absolute;
  right: 0;
  top: 20px;
`;

export const Button = styled.button<{ active?: boolean }>`
  height: 40px;
  width: 40px;
  cursor: pointer;
  background-color: initial;
  border: none;
  opacity: 0.6;
  ${({ active }) =>
    active &&
    css`
      border-bottom: 2px solid #24ffb5;
      opacity: 1;
    `}
`;
