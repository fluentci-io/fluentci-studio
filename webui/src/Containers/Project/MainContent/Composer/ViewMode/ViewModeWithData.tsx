import { FC, useEffect } from "react";
import ViewMode from "./ViewMode";
import { useRecoilState, useRecoilValue } from "recoil";
import { ViewModeState } from "./ViewModeState";
import { TabsState } from "../../TabsState";

const ViewModeWithData: FC = () => {
  const [mode, setMode] = useRecoilState(ViewModeState);
  const tab = useRecoilValue(TabsState);

  useEffect(() => {
    setMode("stacked");
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [tab]);

  return <ViewMode mode={mode} onSetViewMode={setMode} />;
};

export default ViewModeWithData;
