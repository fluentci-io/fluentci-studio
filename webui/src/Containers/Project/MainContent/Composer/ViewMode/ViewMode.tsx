import { FC } from "react";
import { Button, Container } from "./styles";
import { ViewStacked, Code } from "@styled-icons/bootstrap";

export type ViewModeProps = {
  mode: "stacked" | "code";
  onSetViewMode: (viewMode: "stacked" | "code") => void;
};

const ViewMode: FC<ViewModeProps> = (props) => {
  const { mode, onSetViewMode } = props;
  return (
    <Container>
      <Button
        active={mode === "stacked"}
        onClick={() => onSetViewMode("stacked")}
      >
        <ViewStacked size={15} color="#fff" />
      </Button>
      <Button active={mode === "code"} onClick={() => onSetViewMode("code")}>
        <Code size={21} color="#fff" />
      </Button>
    </Container>
  );
};

export default ViewMode;
