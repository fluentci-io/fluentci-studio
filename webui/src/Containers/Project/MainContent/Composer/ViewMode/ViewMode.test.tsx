import { render } from "@testing-library/react";
import ViewMode from "./ViewMode";

describe("ViewMode", () => {
  it("should render", () => {
    const { container } = render(
      <ViewMode mode="stacked" onSetViewMode={vi.fn()} />
    );
    expect(container).toMatchSnapshot();
  });
});
