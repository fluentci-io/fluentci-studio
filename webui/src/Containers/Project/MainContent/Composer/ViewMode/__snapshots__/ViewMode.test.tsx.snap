// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ViewMode > should render 1`] = `
<div>
  <div
    class="css-1rwt18y"
  >
    <button
      class="css-olxsem"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
        color="#fff"
        fill="currentColor"
        focusable="false"
        height="15"
        viewBox="0 0 16 16"
        width="15"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M3 0h10a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2V2a2 2 0 0 1 2-2zm0 1a1 1 0 0 0-1 1v3a1 1 0 0 0 1 1h10a1 1 0 0 0 1-1V2a1 1 0 0 0-1-1H3zm0 8h10a2 2 0 0 1 2 2v3a2 2 0 0 1-2 2H3a2 2 0 0 1-2-2v-3a2 2 0 0 1 2-2zm0 1a1 1 0 0 0-1 1v3a1 1 0 0 0 1 1h10a1 1 0 0 0 1-1v-3a1 1 0 0 0-1-1H3z"
        />
      </svg>
    </button>
    <button
      class="css-ualqix"
    >
      <svg
        aria-hidden="true"
        class="StyledIconBase-sc-ea9ulj-0 gEpWWb"
        color="#fff"
        fill="currentColor"
        focusable="false"
        height="21"
        viewBox="0 0 16 16"
        width="21"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M5.854 4.854a.5.5 0 1 0-.708-.708l-3.5 3.5a.5.5 0 0 0 0 .708l3.5 3.5a.5.5 0 0 0 .708-.708L2.707 8l3.147-3.146zm4.292 0a.5.5 0 0 1 .708-.708l3.5 3.5a.5.5 0 0 1 0 .708l-3.5 3.5a.5.5 0 0 1-.708-.708L13.293 8l-3.147-3.146z"
        />
      </svg>
    </button>
  </div>
</div>
`;
